codepulse/src/main/scala/com/secdec/codepulse/tracer/ProjectUploadData.scala

```diff
@@ -69,6 +69,19 @@ object ProjectUploadData {
 		}
 	}
 
+	def checkForClassesInNestedArchive(file: File): Boolean = {
+		ZipEntryChecker.findFirstEntry(file) { (filename, entry, contents) =>
+			if (!entry.isDirectory) {
+				FilenameUtils.getExtension(entry.getName) match {
+					case "class" => true
+					case _ => false
+				}
+			} else {
+				false
```

Comment:
The predicate passed to `findFirstEntry` here is logically equivalent to the one used in `checkForBinaryZip` (a few lines above this function). (The slight difference is that FilenameUtils will deal with the edge case of a folder named "foo.class", but that doesn't matter since it's in a guard against directories anyway).

You could just use `!entry.isDirectory && FilenameUtils.getExtension(entry.getName) == "class"`